### PR TITLE
magma 2.7.1 2022b/CUDA

### PR DESCRIPTION
**Pull request by fizwit**
(created using `eb --new-pr`)

Magma 2.7.1 add support for CUDA 12.  I have bumped the version to 2.7.1.  magma 2.7.0 fails with CUDA 12 because texture mapping has been removed from CUDA 12. magma-2.7.1 with CUDA 12 builds at my site.
CMake is using the correct compute capability based on EasyBuild config.cfg
```
-- Found CUDA
--     CUDA_CUDART_LIBRARY: CUDA::cudart
--     compile for CUDA arch 6.1 (Pascal)
--     compile for CUDA arch 7.5 (Turing)
```

magma now requires BLAS, toolchain changed to foss.
```
-- Found BLAS: /app/software/FlexiBLAS/3.2.1-GCC-12.2.0/lib/libflexiblas.so
-- Looking for Fortran cheev
-- Looking for Fortran cheev - found
-- Found LAPACK: /app/software/FlexiBLAS/3.2.1-GCC-12.2.0/lib/libflexiblas.so;-lm;-ldl
```

### CODE REVIEW

**Comment by fizwit** [comment]
looks like I should use: UCX-CUDA-1.13.1-GCCcore-12.2.0-CUDA-12.0.0.eb

File: easybuild/easyconfigs/m/magma/magma-2.7.0-GCC-12.2.0-CUDA-12.0.0.eb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ LAPACK but for heterogeneous/hybrid architectures, starting with current Multicore+GPU systems."""
+
+toolchain = {'name': 'GCC', 'version': '12.2.0'}
+toolchainopts = {'pic': True, 'openmp': True}
+
+source_urls = ['https://icl.cs.utk.edu/projectsfiles/%(name)s/downloads/']
+sources = [SOURCE_TAR_GZ]
+checksums = ['fda1cbc4607e77cacd8feb1c0f633c5826ba200a018f647f1c5436975b39fd18']
+
+#patches = ['%(name)s-%(version_major_minor)s.1_allow-all-sms.patch']
```
Comment (Micket) [inline easybuild/easyconfigs/m/magma/magma-2.7.0-GCC-12.2.0-CUDA-12.0.0.eb:18]: This needs to be addressed. I got stuck on this as well, as they have made major changes, it's unclear whether it's needed at all anymore, and if so, needs to be completely rewritten to allow all SM's in the new built procedure.

Comment (fizwit) [reply easybuild/easyconfigs/m/magma/magma-2.7.0-GCC-12.2.0-CUDA-12.0.0.eb:18]: 2.7.0 seems to find the correct GPU targets.
EB is setting this correctly as an argument to CMake: `-DGPU_TARGET="sm_61 sm_75" ` These are the values from my local config.cfg 

CMake output:
```
-- Found CUDA
--     CUDA_CUDART_LIBRARY: CUDA::cudart
--     compile for CUDA arch 6.1 (Pascal)
--     compile for CUDA arch 7.5 (Turing)
```

Comment (Micket) [reply easybuild/easyconfigs/m/magma/magma-2.7.0-GCC-12.2.0-CUDA-12.0.0.eb]: good enough for me, i'll make sure it picks up my options when i test build later

Comment (fizwit) [reply easybuild/easyconfigs/m/magma/magma-2.7.0-GCC-12.2.0-CUDA-12.0.0.eb]: should remove the patches line?

**Review by Micket — APPROVED** [review]
lgtm

> magma now requires BLAS, toolchain changed to foss.

This really confused me. It's always been at foss level, and BLAS dep isn't something new.

**Comment by Micket** [comment]
@boegelbot please test @ jsc-zen2

**Pull request merged**